director/background_tasks.go

```diff
@@ -140,7 +140,7 @@ func pushAll() error {
 			}
 		}
 		// This contrived bit of logic is to handle devices that don't have a LastScheduledPush set yet
-		if !dbDevice.LastScheduledPush.Before(lastCheckinDelay) {
+		if !dbDevice.LastScheduledPush.IsZero() || dbDevice.LastCertificateList.IsZero() || dbDevice.LastProfileList.IsZero() || dbDevice.LastSecurityInfo.IsZero() || dbDevice.LastDeviceInfo.IsZero() {
```

Comment:
LastScheduledPush is time.Time and IsZero is nil value for time?

Reply:
Yes - i.e. has one ever happened?